internal/backend/backend_retry.go

```diff
@@ -33,6 +33,10 @@ func NewRetryBackend(be restic.Backend, maxTries int, report func(string, error,
 }
 
 func (be *RetryBackend) retry(ctx context.Context, msg string, f func() error) error {
+	if ctx.Err() != nil {
+		return ctx.Err()
+	}
```

Comment:
Since this check isn't strictly necessary (assuming backends work properly; backoff.RetryNotify also checks the context), I think it should have a comment.

Reply:
I've added a few lines of comments. However, I'm not sure whether that covers the `assuming backends work properly` part of your comment. Does "working properly" necessaryly entail checking whether a context is cancelled, even if the backend uses no cancelable functions? The context check by `RetryNotify` only happens between retries, so the first try is not covered.

Reply:
That is what I meant, but thinking about it more, I'm no longer sure it's the right definition of "working"...

Reply:
I've added a sentence which mentions that the local backend doesn't prevent operations from working once a context is canceled.

> That is what I meant, but thinking about it more, I'm no longer sure it's the right definition of "working"...

My expectation was/is that the operation fails when the context was already canceled. After all, why would we pass a context to the method otherwise?